cmd/vault-login-oauth/vault-login-oauth.go

```diff
@@ -0,0 +1,261 @@
+// vault-login-oauth is the program a user runs to log in to Vault with OAuth.
+//
+// It will:
+//
+// 1. Query Vault to determine what the authorization request URL should be,
+//    according to the configuration there.
```

Comment:
I wonder if this is a bad idea -- how sure can we be that we're receiving authentic configuration information? Ostensibly the user would notice if they end up sent to https://accounts.gooogle.com/ and not enter a password there, but in practice probably not.